src/session.rs

```diff
@@ -117,6 +122,13 @@ impl Session {
         );
 
         let response_bytes = response.encode()?;
+
+        record_request_duration(
+            &self.config.chain_id,
+            &request_type,
+            request_start.elapsed(),
+        );
```

Comment:
if i see correctly, there are several points where the function can early return

and when it does, no time is recorded

also, IIUC, the metrics will not be reset if an error happens, so the metric will be "stuck" at the same value.
It's an all time "problem" of how prometheus client thinks we should export metrics, so it's just FYI.